### PR TITLE
Added cmake3 binary name to search for external

### PR DESCRIPTION
**Pull request by OliverPerks**
Support for search for 'cmake' and 'cmake3' (as we see it in yum based OSs)

### CODE REVIEW

File: var/spack/repos/builtin/packages/cmake/package.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ class Cmake(Package):
 
     tags = ['build-tools']
 
-    executables = ['^cmake$']
+    executables = ['^cmake$', '^cmake3$']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cmake/package.py]:
```suggestion
    executables = ['^cmake[0-9]*$']
```
Wouldn't this be more robust (e.g., cmake 2.8, 4.0, etc.)?

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
Can you resolve the merge conflicts?

**Comment by alalazo** [comment]
Ping @OliverPerks

**Review by bernhardkaindl — DISMISSED** [review]
Only needed for RHEL7/CentOS7 because RHEL8 make cmake3 the default. But as RHEL7 is still maintained, lets merge it.

**Comment by bernhardkaindl** [comment]
@spackbut fix style

**Comment by bernhardkaindl** [comment]
@spackbot fix style

**Review by bernhardkaindl — APPROVED** [review]
Suggestion by adamsteward was integrated and conflict with develop resolved.

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by adamjstewart — APPROVED** [review]

**Comment by alalazo** [comment]
Force merged, since this one cannot make a pipeline build fail that previously wasn't

**Comment by Jordan474** [comment]
This actually broke bootstrap on rhel7 for me: because it detects `cmake3` as available external, but then the clingo package runs `cmake`, which is cmake 2.

Many other packages could be affected by this when one does `spack external find`.

I guess one could change `Cmake.setup_dependent_package` to set the the `cmake` module function to `bin/cmake3`, but I'm not sure how robust this will be, and if this will cover all cmake usages. I would argue it is safer to ignore `cmake3` and only detect "vanilla" cmake installation `bin/cmake` ?

**Comment by sebrowne** [comment]
> This actually broke bootstrap on rhel7 for me: because it detects `cmake3` as available external, but then the clingo package runs `cmake`, which is cmake 2.
> 
> Many other packages could be affected by this when one does `spack external find`.
> 
> I guess one could change `Cmake.setup_dependent_package` to set the the `cmake` module function to `bin/cmake3`, but I'm not sure how robust this will be, and if this will cover all cmake usages. I would argue it is safer to ignore `cmake3` and only detect "vanilla" cmake installation `bin/cmake` ?

We encountered this as well at our site.

**Pull request merged**